In [2]:
%cd ..

c:\Users\calag\Desktop\nlp-proyecto7-pc3


In [6]:
from src.vision_encoder import VisionEncoder
from torchvision import transforms
from PIL import Image
import torch

In [9]:
#MI IMAGEN DE PRUEBA
img = Image.open("notebooks/dibujo_mio.png").convert("RGB")

#TRANSFORMACIÓN QUE ViT ESPERA
transform = transforms.Compose([
    transforms.Resize((224, 224)),
    transforms.ToTensor(),
    transforms.Normalize(mean=[0.5]*3, std=[0.5]*3)
])

img_tensor = transform(img).unsqueeze(0)  # (1, 3, 224, 224)

device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
encoder = VisionEncoder(device=device)

with torch.no_grad():
    output = encoder(img_tensor.to(device))

print("Shape del embedding visual:", output.shape)

Downloading: "https://download.pytorch.org/models/vit_b_16-c867db91.pth" to C:\Users\calag/.cache\torch\hub\checkpoints\vit_b_16-c867db91.pth


100.0%


Shape del embedding visual: torch.Size([1, 768])


Prueba básica para "text_encoder.py"

In [3]:
from src.text_encoder import TextEncoder
import torch

device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
encoder = TextEncoder(device=device)

texts = ["a dog playing in the grass", "a person riding a bicycle"]

embeddings = encoder(texts)
print("Shape de los embeddings de texto:", embeddings.shape)  # (B, L, D)


c:\Users\calag\Desktop\nlp-proyecto7-pc3\venv\Lib\site-packages\tqdm\auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm
c:\Users\calag\Desktop\nlp-proyecto7-pc3\venv\Lib\site-packages\huggingface_hub\file_download.py:143: UserWarning: `huggingface_hub` cache-system uses symlinks by default to efficiently store duplicated files but your machine does not support them in C:\Users\calag\.cache\huggingface\hub\models--distilbert-base-uncased. Caching files will still work but in a degraded version that might require more space on your disk. This warning can be disabled by setting the `HF_HUB_DISABLE_SYMLINKS_WARNING` environment variable. For more details, see https://huggingface.co/docs/huggingface_hub/how-to-cache#limitations.
To support symlinks on Windows, you either need to activate Developer Mode or to run Python as an administrator. In o

Shape de los embeddings de texto: torch.Size([2, 8, 768])


Esto devuelve embeddings con tamaño:

B: número de frases

L: longitud máxima de tokens por frase

D: dimensión del embedding de DistilBERT (768)